priam/src/main/java/com/netflix/priam/resources/BackupServletV2.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.priam.resources;
+
+import com.google.inject.Inject;
+import com.netflix.priam.backup.BackupMetadata;
+import com.netflix.priam.backup.BackupVerification;
+import com.netflix.priam.backup.BackupVerificationResult;
+import com.netflix.priam.backup.IBackupStatusMgr;
+import com.netflix.priam.services.BackupTTLService;
+import com.netflix.priam.services.SnapshotMetaService;
+import com.netflix.priam.utils.DateUtil;
+import com.netflix.priam.utils.DateUtil.DateRange;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Created by aagrawal on 1/16/19. */
+@Path("/v2/backup")
+@Produces(MediaType.APPLICATION_JSON)
+public class BackupServletV2 {
+    private static final Logger logger = LoggerFactory.getLogger(BackupServletV2.class);
+    private final BackupVerification backupVerification;
+    private final IBackupStatusMgr backupStatusMgr;
+    private final SnapshotMetaService snapshotMetaService;
+    private final BackupTTLService backupTTLService;
+    private static final String REST_SUCCESS = "[\"ok\"]";
+
+    @Inject
+    public BackupServletV2(
+            IBackupStatusMgr backupStatusMgr,
+            BackupVerification backupVerification,
+            SnapshotMetaService snapshotMetaService,
+            BackupTTLService backupTTLService) {
+        this.backupStatusMgr = backupStatusMgr;
+        this.backupVerification = backupVerification;
+        this.snapshotMetaService = snapshotMetaService;
+        this.backupTTLService = backupTTLService;
+    }
+
+    @GET
+    @Path("/do_snapshot")
+    public Response backup() throws Exception {
+        snapshotMetaService.execute();
+        return Response.ok(REST_SUCCESS, MediaType.APPLICATION_JSON).build();
+    }
+
+    @GET
+    @Path("/ttl")
+    public Response ttl() throws Exception {
+        backupTTLService.execute();
+        return Response.ok(REST_SUCCESS, MediaType.APPLICATION_JSON).build();
+    }
+
+    @GET
+    @Path("/info/{date}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response info(@PathParam("date") String date) throws Exception {
+        Instant instant = DateUtil.parseInstant(date);
+        List<BackupMetadata> metadataList =
+                backupStatusMgr.getLatestBackupMetadata(
+                        SnapshotMetaService.BACKUP_VERSION,
+                        new DateRange(
+                                instant,
+                                instant.plus(1, ChronoUnit.DAYS).truncatedTo(ChronoUnit.DAYS)));
+        return Response.ok(metadataList.toString(), MediaType.APPLICATION_JSON).build();
```

Comment:
Do we want the `toString`? Without it we'll get a nice list of objects instead of a string that contains a serialized object right?